src/analysis/convex/basic.lean

```diff
@@ -29,207 +28,388 @@ We also provide various equivalent versions of the definitions above, prove that
 are convex, and prove Jensen's inequality.
 
 Note: To define convexity for functions `f : E → β`, we need `β` to be an ordered vector space,
-defined using the instance `ordered_smul ℝ β`.
+defined using the instance `ordered_smul 𝕜 β`.
 
 ## Notations
 
-We use the following local notations:
-
-* `I = Icc (0:ℝ) 1`;
-* `[x, y] = segment x y`.
+We use the following notations:
+* `[x -[𝕜] y] = segment 𝕜 x y`
+* `I = Icc (0 : ℝ) 1`
```

Comment:
Patrick requested that `[x, y]` was still available with the older meaning. Can we have that?

Reply:
It was local notation, so we never "had" it (or can local notation be used outside of the file defining it using namespaces or such?). But. And of course we can't have it a global notation because that would conflict with lists. I'm for defining it where it's used or putting it in the `real` locale (how does one do that?).